src/injector/src/main.rs

```diff
@@ -100,77 +102,91 @@ fn start_seed_registry() {
-                        },
-                        "*/*" => Response::empty_406()
-                    )
-                },
-
-                (HEAD) (/v2/{_namespace :String}/registry/manifests/{_ref :String}) => {
-                    // a normal HEAD response has an empty body, but due to rouille not allowing for an override
-                    // on Content-Length, we respond the same as a GET
-                    accept!(
-                        request,
-                        "application/vnd.docker.distribution.manifest.v2+json" => {
-                            handle_get_manifest(&root)
-                        },
-                        "*/*" => Response::empty_406()
-                    )
-                },
-
-                (GET) (/v2/registry/blobs/{digest :String}) => {
-                    handle_get_digest(&root, &digest)
-                },
-
-                (GET) (/v2/{_namespace :String}/registry/blobs/{digest :String}) => {
-                    handle_get_digest(&root, &digest)
+                        .with_unique_header("Content-Type", "application/json; charset=utf-8")
+                        .with_additional_header("Docker-Distribution-Api-Version", "registry/2.0")
+                        .with_additional_header("X-Content-Type-Options", "nosniff")
                 },
 
                 _ => {
-                    Response::empty_404()
+                    handle_request(&root, &request)
                 }
             )
         })
     });
 }
 
+fn handle_request(root: &Path, request: &Request) -> Response {
+    let url = request.url();
+    let url_segments: Vec<_> = url.split("/").collect();
+    let url_seg_len = url_segments.len();
+
+    if url_seg_len >= 4 && url_segments[1] == "v2" {
+        let tag_index = url_seg_len - 1;
+        let object_index = url_seg_len - 2;
+
+        let object_type = url_segments[object_index];
+
+        if object_type == "manifests" {
+            let tag_or_digest = url_segments[tag_index].to_owned();
+
+            let namespaced_name = url_segments[2..object_index].join("/");
+
+            if request.method() == "GET" {
+                return handle_get_manifest(&root, &namespaced_name, &tag_or_digest);
+            } else if request.method() == "HEAD" {
+                // a normal HEAD response has an empty body, but due to rouille not allowing for an override
+                // on Content-Length, we respond the same as a GET
+                return accept!(
+                    request,
+                    DOCKER_MIME_TYPE => {
+                        handle_get_manifest(&root, &namespaced_name, &tag_or_digest)
+                    },
+                    "*/*" => Response::empty_406()
+                );
+            }
+        } else if object_type == "blobs" && request.method() == "GET" {
+            let digest = url_segments[tag_index].to_owned();
+            return handle_get_digest(&root, &digest);
+        }
+    }
+
+    Response::empty_404()
+}
```

Comment:
Can you add comments that specify which code is handling which route ie:

```
// this route handles (GET) /v2/**/manifests/<reference>
```

Reply:
done